app.py

```
@@ -571,42 +758,124 @@ def view_complaints():
+    try:
+        if attachment:
+            file_path, stored_name = save_media_file(attachment)
+            file_name = attachment.filename or stored_name
+            media_mime_type = attachment.mimetype or mimetypes.guess_type(file_path.name)[0] or 'application/octet-stream'
+            message_type = detect_message_type(media_mime_type)
+            media_id = upload_media_to_whatsapp(file_path, media_mime_type)
+            send_whatsapp_message(mobile, message_type, text=message_text or None, media_id=media_id, file_name=file_name)
+            media_url = f"uploads/whatsapp/{stored_name}"
+        else:
+            send_whatsapp_message(mobile, 'text', text=message_text)
+    except Exception as exc:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)